08-components-lab/src/02-numberguessing.jsx

```diff
@@ -1,17 +1,66 @@
-/**
- * Lab 02:
- *
- * Implement a number guessing game component.
- *
- * The component should provide an input field for the user to guess a
- * number picked randomly by the computer (from 1 to 1,000).
- * When typing a number, the component should print "too low" or "too high" 
- * as a hint to the user.
- *
- * To make the game interesting, computer should sometimes cheat and print wrong messages.
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+/**
+ * Lab 02:
+ *
+ * Implement a number guessing game component.
+ *
+ * The component should provide an input field for the user to guess a
+ * number picked randomly by the computer (from 1 to 1,000).
+ * When typing a number, the component should print "too low" or "too high" 
+ * as a hint to the user.
+ *
+ * To make the game interesting, computer should sometimes cheat and print wrong messages.
+ */
+
+import React from 'react';
+
+export default class MultiInput extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            result: "start guessing...",
+            number: Math.floor(Math.random() * (1000 - 1)) + 1,
+        };
+    }
+
+    guess = (ev) => {
+        console.log("guessed " + ev.target.value);
+        var result = this.checkResult(ev.target.value);
+        console.log(result);
+        this.setState({ result: result });
+    };
+
+    checkResult = (number) => {
+        var liar = Math.floor(Math.random() * (10 - 0)) + 0;

```

Comment:
Also note that you can destructure number from state by using
const { number } = this.state;